contrib/src/main/java/com/datatorrent/contrib/enrich/FSLoader.java

```diff
@@ -145,9 +136,44 @@ private Object getKey(Map<String, Object> tuple)
   @Override
   public Object get(Object key)
   {
+    try (
+      FSDataInputStream in = openFsInputStream(filePath);
+      BufferedReader bin = new BufferedReader(new InputStreamReader(in));
+      ) {
+      String line;
+      while ((line = bin.readLine()) != null) {
+        try {
+          Map<String, Object> tuple = extractFields(line);
+          if (tuple != null && !tuple.isEmpty() && getKey(tuple).equals(key)) {
+            logger.debug("Found line in FS {}", getValue(tuple));
+            return getValue(tuple);
+          }
+        } catch (Exception parseExp) {
+          logger.warn("Unable to parse line {}", line);
+        }
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
     return null;
   }
 
+  private FSDataInputStream openFsInputStream(Path filePath)
```

Comment:
Maybe call it check open or something that indicates it is not a mere open but will try to handle failures and retry